src/stratigraphy/util/find_description.py

```diff
@@ -48,30 +50,30 @@ def get_description_blocks(
     if len(distances):
         threshold = min(distances) * 1.15
 
-    blocks = []
-    current_block_lines = []
-    for line in description_lines:
-        if len(current_block_lines) > 0:
-            last_line = current_block_lines[-1]
-            if _block_separated_by_line(
-                last_line, line, TextBlock(current_block_lines), geometric_lines, threshold=block_line_ratio
-            ):
-                blocks.append(TextBlock(current_block_lines, is_terminated_by_line=True))
-                current_block_lines = []
-            elif _block_separated_by_lefthandside_line_segment(
-                last_line,
-                line,
-                TextBlock(current_block_lines),
+    # Create blocks separated by lines
+    blocks = _create_blocks_by_separator(
+        description_lines,
+        geometric_lines,
+        _block_separated_by_line,
+        {"threshold": block_line_ratio},
+        set_terminated_by_line_flag=True,
+    )
+
+    # Create blocks separated by lefthandside line segments
+    _blocks = []
+    for block in blocks:
+        _blocks.extend(
+            _create_blocks_by_separator(
+                block.lines,
                 geometric_lines,
-                length_threshold=left_line_length_threshold,
-            ):
-                blocks.append(
-                    TextBlock(current_block_lines, is_terminated_by_line=False)
-                )  # we consider a splitting due to a lefthandside line segment as weak.
-                current_block_lines = []
-        current_block_lines.append(line)
-    if len(current_block_lines):
-        blocks.append(TextBlock(current_block_lines))
+                _block_separated_by_lefthandside_line_segment,
```

Comment:
Maybe pass this as e.g.
`lambda last_line, line, block: _block_separated_by_lefthandside_line_segment(last_line, line, block, geometric_lines, left_line_length_threshold)`? Then the method `_create_blocks_by_separator` itself can do with two fewer parameters.

Reply:
I personally find the code clearer to read and understand the way it is currently. I agree that we'd need to pass fewer arguments in your case, which comes with the benefit of added usage.

If we are not to use this function very frequently in other places too, I would keep it. But happy to implement otherwise if you prefer.